Source/Particles/NamedComponentParticleContainer.H

```diff
@@ -16,12 +16,19 @@
 #include <map>
 #include <string>
 
-/** Particle Attributes stored in amrex::ParticleContainer's struct of array
+/** Real Particle Attributes stored in amrex::ParticleContainer's struct of array
  */
 struct PIdx
 {
     enum {
-        w = 0,      ///< weight
+#if !defined (WARPX_DIM_1D)
+        x,
+#endif
+#if defined (WARPX_DIM_3D)
+        y,
+#endif
+        z,
+        w,      ///< weight
         ux, uy, uz,
 #ifdef WARPX_DIM_RZ
         theta,      ///< RZ needs all three position components
```

Comment:
We could move this forward in the enum to `r,z,theta`.

Reply:
And even better, we should make `theta` a runtime parameter (see reason in https://github.com/AMReX-Codes/pyamrex/pull/243)